frontend/src/pages/hardwareProfiles/useHardwareProfilesByFeatureVisibility.ts

```diff
@@ -1,23 +1,37 @@
 import React from 'react';
 import { HardwareProfileKind, HardwareProfileFeatureVisibility } from '#~/k8sTypes';
 import { isHardwareProfileValid } from '#~/pages/hardwareProfiles/utils';
-import { ProjectDetailsContext } from '#~/pages/projects/ProjectDetailsContext';
 import { HardwareProfilesContext } from '#~/concepts/hardwareProfiles/HardwareProfilesContext';
+import { ProjectDetailsContext } from '#~/pages/projects/ProjectDetailsContext.tsx';
+import { CustomWatchK8sResult } from '#~/types.ts';
 
 export const useHardwareProfilesByFeatureVisibility = (
   visibility?: HardwareProfileFeatureVisibility[],
+  projectHardwareProfiles?: CustomWatchK8sResult<HardwareProfileKind[]>,
 ): {
   projectProfiles: [data: HardwareProfileKind[], loaded: boolean, loadError: Error | undefined];
   globalProfiles: [data: HardwareProfileKind[], loaded: boolean, loadError: Error | undefined];
 } => {
   const {
-    globalHardwareProfiles: [globalProfiles, globalLoaded, globalError],
+    globalHardwareProfiles: [globalProfiles, globalProfilesLoaded, globalProfilesError],
   } = React.useContext(HardwareProfilesContext);
   const {
     currentProject,
-    projectHardwareProfiles: [projectProfiles, projectLoaded, projectError],
+    projectHardwareProfiles: [
+      projectProfilesFromContext,
+      projectProfilesFromContextLoaded,
+      projectProfilesFromContextError,
+    ],
   } = React.useContext(ProjectDetailsContext);
 
+  const [givenProjectProfiles, givenProjectProfilesLoaded, givenProjectProfilesError] =
+    projectHardwareProfiles || [];
+  const projectProfiles = givenProjectProfiles || projectProfilesFromContext;
+  const projectProfilesLoaded =
+    givenProjectProfilesLoaded ||
+    (currentProject.metadata.name ? projectProfilesFromContextLoaded : true);
+  const projectProfilesError = givenProjectProfilesError || projectProfilesFromContextError;
+
```

Comment:
When we start to find the prop drilldowns like this, it's time to take a step back to rethink the architecture because there are so many overlaps with the `projectProfilesFromContext` and `givenProjectProfiles`. And if you trace them down, they are actually fetched mostly in the same way and use the same data, we only need the minimum data we need, and let the hook finish the logic perfectly. The callers will never need to worry about getting the extra hardware profiles and passing it into this hook then, and they never should.